File: single.responsibility.principles.example.js
function Square(length) {
    this.length = length;
}

function Circle(radius) {
    this.radius = radius;
}

function Caculator(shapes) {
    this.shapes = shapes;
}

Caculator.prototype.sum = function () {
    areas = [];
    
    shapes.forEach((shape) => {        
        if ( shape instanceof Square) {
            areas.push(Math.pow(shape.length, 2))
        } else if (shape instanceof Circle) {
            areas.push(Math.PI*Math.pow(shape.radius, 2))
        }
    });
    
    let sum = 0.0
    sum = areas.reduce((a,b) => a + b, 0);
    
    return sum;
}

let shapes = [new Circle(2), new Square(5), new Square(6)];
console.log(shapes);

let areas = new Caculator(shapes);
console.log(areas);

console.log("this is the sum():" + areas.sum());

File: single.responsibility.principles.example.java
import java.lang.Math.*;
import java.util.*;

public class HelloWorld{
    
    public static class Square
    {
        public Integer length;
        
        public Square(Integer length)
        {
            this.length = length;
        }
    }
    
    public static class Circle
    {
        public Integer radius;
        
        public Circle(Integer radius)
        {
            this.radius = radius;
            System.out.println("radius of a new Circle is: " + this.radius);
        }
    }
    
    public static class Caculator
    {
        public Object shapes[];
        
        //List<Double> areas = new ArrayList<Double> ();  output will run sum() again, and areas will have duplicate objects, unless it's been created in sum() scope. 
        
        public Caculator(Object shapes[])
        {
            this.shapes = shapes;
            //System.out.println(this.shapes);
        }
        
        public String sum()
        {
            List<Double> areas = new ArrayList<Double> (); 
            
            for(Object shape : this.shapes){
                if(shape instanceof Square) {
                    Square s = (Square)shape;
                    areas.add(Math.pow(s.length, 2));
                }  else if (shape instanceof Circle) {
                    Circle c = (Circle)shape;
                    areas.add(Math.PI*Math.pow(c.radius, 2));
                }
            }
            
            Double sum = 0.0;
            sum = areas.stream()
            .mapToDouble(a -> a)
            .sum();

            return sum.toString(); 

        }
        
        public String output()
        {
            return "Sum of the areas of provided shapes: " + this.sum();
        } 
    } 

    public static void main(String []args){
        Object shapes[] = {
            new Circle(2),
            new Square(5),
            new Square(6),
        };
        
        Caculator areas = new Caculator(shapes);

        System.out.println("this is the output(): " + areas.output());
    }
    
}
